### PR TITLE
[orchagent]: Create the switch with source MAC address

### PR DESCRIPTION
**Pull request by stcheng**
- Instead of creating the switch and later setting the source MAC address,
  use the source MAC address as an attribute to create the switch.

- Move corresponding switch attribute code to above the switch creation.

### CODE REVIEW

File: orchagent/main.cpp [hunk 1]
```diff
@@ -227,61 +206,73 @@ int main(int argc, char **argv)
...
         if (status != SAI_STATUS_SUCCESS)
         {
-            SWSS_LOG_ERROR("Failed to get MAC address from switch %d", status);
+            SWSS_LOG_ERROR("Failed to get MAC address from switch, rv:%d", status);
             exit(EXIT_FAILURE);
         }
         else
         {
             gMacAddress = attr.value.mac;
         }
```
Comment (lguohan) [inline orchagent/main.cpp]: you do not have switchid yet, you need to move this after create_switch.

**Comment by lguohan** [comment]
>         status = sai_switch_api->set_switch_attribute(gSwitchId, &attr);

you need to create_switch first.

---
Refers to: orchagent/main.cpp:179 in 75806d0. [](commit_id = 75806d08fe9fbb067d33523d485a9ea23b330903, deletion_comment = False)

**Comment by stcheng** [comment]
>         status = sai_switch_api->set_switch_attribute(gSwitchId, &attr);

this is redis attributes, no need to create the switch first. please see above comments by Kamil.

---
In reply to: [315475884](https://github.com/Azure/sonic-swss/pull/253#issuecomment-315475884) [](ancestors = 315475884)

---
Refers to: orchagent/main.cpp:179 in 75806d0. [](commit_id = 75806d08fe9fbb067d33523d485a9ea23b330903, deletion_comment = False)

File: orchagent/main.cpp (hunk 1 above)
Comment (stcheng) [inline orchagent/main.cpp]: fixed.

---
In reply to: [127554898](https://github.com/Azure/sonic-swss/pull/253#discussion_r127554898) [](ancestors = 127554898)

**Comment by lguohan** [comment]
>         status = sai_switch_api->set_switch_attribute(gSwitchId, &attr);

should mark as by design. won't fix means acknowledge it is a bug.

---
In reply to: [315479635](https://github.com/Azure/sonic-swss/pull/253#issuecomment-315479635) [](ancestors = 315479635,315475884)

---
Refers to: orchagent/main.cpp:179 in 75806d0. [](commit_id = 75806d08fe9fbb067d33523d485a9ea23b330903, deletion_comment = False)

**Pull request merged**